From hackermail Thu Jan  1 00:00:00 1970
From: SeongJae Park <sj@kernel.org>
To: Andrew Morton <akpm@linux-foundation.org>
Cc: SeongJae Park <sj@kernel.org>
Cc: damon@lists.linux.dev
Cc: kernel-team@meta.com
Cc: linux-kernel@vger.kernel.org
Cc: linux-mm@kvack.org
Date: Sat, 5 Jul 2025 14:57:21 -0700
Subject: [PATCH mm-unstable] mm/damon/stat: reset enabled when DAMON start
         failed

When damon_stat_start() fails, 'enabled' parameter is not reset.  As a
result, users show the parameter 'Y' while it is not really working.
Fix it by resetting 'enabled' to 'false' when damon_stat_start() failed.

Fixes: 00e74895aba3 ("mm/damon: introduce DAMON_STAT module") # mm-unstable
Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/stat.c | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/mm/damon/stat.c b/mm/damon/stat.c
index c35bd1852f0b..b75af871627e 100644
--- a/mm/damon/stat.c
+++ b/mm/damon/stat.c
@@ -225,8 +225,12 @@ static int damon_stat_enabled_store(
 		 */
 		return 0;
 
-	if (enabled)
-		return damon_stat_start();
+	if (enabled) {
+		err = damon_stat_start();
+		if (err)
+			enabled = false;
+		return err;
+	}
 	damon_stat_stop();
 	return 0;
 }
@@ -240,6 +244,9 @@ static int __init damon_stat_init(void)
 	/* probably set via command line */
 	if (enabled)
 		err = damon_stat_start();
+
+	if (err && enabled)
+		enabled = false;
 	return err;
 }
 

base-commit: 93a51997eb1dda993197a9f82602b96d56056ce4
-- 
2.39.5
